scripts/release/cabal-plan-diff.sh

```diff
@@ -72,82 +72,81 @@ cabal build all --dry-run --minimize-conflict-set --project-file='./tmp/cabal.pr
+
+  awk '/^$/{f=!f} f' ./tmp/the-diff-1 > ./tmp/the-diff
+
+  grep "^\-" ./tmp/the-diff | cut -d'-' -f2- | cut -d' ' -f1 > ./tmp/only-in-full
+  grep "^\+" ./tmp/the-diff | cut -d'+' -f2- | cut -d' ' -f1 > ./tmp/only-in-libs
+
+  echo -e "\n=== TEST ONLY DEPENDENCIES"
+
+  comm -2 -3 <(rev ./tmp/only-in-full | cut -d'-' -f2- | rev | uniq) <(rev ./tmp/only-in-libs | cut -d'-' -f2- | rev | uniq)
+
+  echo -e "\n=== FULL-BUILD ONLY DEPENDENCIES"
+
+  comm -1 -3 <(rev ./tmp/only-in-full | cut -d'-' -f2- | rev | uniq) <(rev ./tmp/only-in-libs | cut -d'-' -f2- | rev | uniq)
+
+  echo -e "\n=== VERSION BUMPS FROM TESTS ONLY TO FULL-BUILD"
+
+  common=$(comm -1 -2 <(rev ./tmp/only-in-full | cut -d'-' -f2- | rev | uniq) <(rev ./tmp/only-in-libs | cut -d'-' -f2- | rev | uniq))
+
+  major=()
+  minor=()
+  patch=()
+
+  for dep in $common; do
+    full=$(grep -E "^$dep-([0-9]+\.)*[0-9]" ./tmp/only-in-full | rev | cut -d'-' -f1 | rev)
+    if [[ $(echo $full | awk -F. '{print NF-1}') -lt 3 ]]; then
+      n=$(((3 - $(echo $full | awk -F. '{print NF-1}'))*2))
+      sff=".0.0.0.0"
+      full="$full${sff:0:$n}"
+    fi
+    libs=$(grep -E "^$dep-([0-9]+\.)*[0-9]" ./tmp/only-in-libs | rev | cut -d'-' -f1 | rev)
+    if [[ $(echo $libs | awk -F. '{print NF-1}') -lt 3 ]]; then
+      n=$(((3 - $(echo $libs | awk -F. '{print NF-1}'))*2))
+      sff=".0.0.0.0"
+      libs="$libs${sff:0:$n}"
+    fi
+    verDiff=$(printf "%s\n" "$full" "$libs" | sed -e 'N;s/^\(.*\).*\n\1.*$/\1/' | awk -F. '{print NF-1}')
+    case $verDiff in
+      0)
+        major+=("$dep $full -> $libs")
+        ;;
+      1)
+        major+=("$dep $full -> $libs")
+        ;;
+      2)
+        minor+=("$dep $full -> $libs")
+        ;;
+      *)
+        patch+=("$dep $full -> $libs")
+        ;;
+    esac
+  done
+
+  echo -e "- MAJOR BUMPS"
+  printf '%s\n' "${major[@]}" | sed 's/^\(.\+\)/   - \1/g'
+
+  echo -e "- MINOR BUMPS"
+  printf '%s\n' "${minor[@]}" | sed 's/^\(.\+\)/   - \1/g'
+
+  echo -e "- PATCH BUMPS"
+  printf '%s\n' "${patch[@]}" | sed 's/^\(.\+\)/   - \1/g'
+
+  if [[ -n $GH && ${#major[@]} ]]; then
+    str=$(printf '%s\n' "${major[@]}")
+    echo "::warning title=Major bumps detected::$str"
   fi
-  verDiff=$(printf "%s\n" "$full" "$libs" | sed -e 'N;s/^\(.*\).*\n\1.*$/\1/' | awk -F. '{print NF-1}')
-  case $verDiff in
-    0)
-      major+=("$dep $full -> $libs")
-      ;;
-    1)
-      major+=("$dep $full -> $libs")
-      ;;
-    2)
-      minor+=("$dep $full -> $libs")
-      ;;
-    *)
-      patch+=("$dep $full -> $libs")
-      ;;
-  esac
-done
-
-echo -e "- MAJOR BUMPS"
-printf '%s\n' "${major[@]}" | sed 's/^\(.\+\)/   - \1/g'
-
-echo -e "- MINOR BUMPS"
-printf '%s\n' "${minor[@]}" | sed 's/^\(.\+\)/   - \1/g'
-
-echo -e "- PATCH BUMPS"
-printf '%s\n' "${patch[@]}" | sed 's/^\(.\+\)/   - \1/g'
-
-if [[ -n $GH && ${#major[@]} ]]; then
-  str=$(printf '%s\n' "${major[@]}")
-  echo "::warning title=Major bumps detected::$str"
-fi
 
-rm -r ./tmp
+  rm -r ./tmp
```

Comment:
This cleanup should be outside of the `if-else` statements

Reply:
Good point

Reply:
Implemented with `trap`